relational_operators/WorkOrderFactory.cpp

```diff
@@ -0,0 +1,602 @@
+    }
+    case serialization::TABLE_GENERATOR: {
+      return proto.HasExtension(serialization::TableGeneratorWorkOrder::generator_function_index) &&
+             query_context.isValidGeneratorFunctionId(
+                 proto.GetExtension(serialization::TableGeneratorWorkOrder::generator_function_index)) &&
+             proto.HasExtension(serialization::TableGeneratorWorkOrder::insert_destination_index) &&
+             query_context.isValidInsertDestinationId(
+                 proto.GetExtension(serialization::TableGeneratorWorkOrder::insert_destination_index));
+    }
+    case serialization::TEXT_SCAN: {
+      if (!proto.HasExtension(serialization::TextScanWorkOrder::field_terminator) ||
+          !proto.HasExtension(serialization::TextScanWorkOrder::process_escape_sequences) ||
+          !proto.HasExtension(serialization::TextScanWorkOrder::insert_destination_index) ||
+          !query_context.isValidInsertDestinationId(
+              proto.GetExtension(serialization::TextScanWorkOrder::insert_destination_index))) {
+        return false;
+      }
+
+      // Two fields are exclusive.
+      if (proto.HasExtension(serialization::TextScanWorkOrder::filename) ==
+              proto.HasExtension(serialization::TextScanWorkOrder::text_blob)) {
+        return false;
+      }
+
+      return proto.HasExtension(serialization::TextScanWorkOrder::filename) ||
+             proto.GetExtension(serialization::TextScanWorkOrder::text_blob).IsInitialized();
+    }
+    case serialization::TEXT_SPLIT: {
+      return proto.HasExtension(serialization::TextSplitWorkOrder::filename) &&
+             proto.HasExtension(serialization::TextSplitWorkOrder::process_escape_sequences) &&
+             proto.HasExtension(serialization::TextSplitWorkOrder::operator_index);
+    }
+    case serialization::UPDATE: {
+      return proto.HasExtension(serialization::UpdateWorkOrder::relation_id) &&
+             catalog_database.hasRelationWithId(
+                 proto.GetExtension(serialization::UpdateWorkOrder::relation_id)) &&
+             proto.HasExtension(serialization::UpdateWorkOrder::insert_destination_index) &&
+             query_context.isValidInsertDestinationId(
+                 proto.GetExtension(serialization::UpdateWorkOrder::insert_destination_index)) &&
+             proto.HasExtension(serialization::UpdateWorkOrder::predicate_index) &&
+             query_context.isValidPredicate(
+                 proto.GetExtension(serialization::UpdateWorkOrder::predicate_index)) &&
+             proto.HasExtension(serialization::UpdateWorkOrder::update_group_index) &&
+             query_context.isValidUpdateGroupId(
+                 proto.GetExtension(serialization::UpdateWorkOrder::update_group_index)) &&
+             proto.HasExtension(serialization::UpdateWorkOrder::operator_index) &&
+             proto.HasExtension(serialization::UpdateWorkOrder::block_id);
+    }
+    default:
+      return false;
+  }
```

Comment:
I was wonder why it is hard to read. I understand that there are too many conditions, but that's because some `WorkOrder` proto has the same number of fields to be serialized.

One way to check the correctness is to compare `WorkOrder` proto and its factory reconstruction method, though.